cmd/upgradehelper/vmlivemigratedetector.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/harvester/harvester/pkg/upgradehelper/vmlivemigratedetector"
+)
+
+var (
+	shutdown bool
+)
+
+var vmLiveMigrateDetectorCmd = &cobra.Command{
+	Use:   "vm-live-migrate-detector NODENAME",
+	Short: "VM Live Migrate Detector",
+	Long: `A simple VM detector and executor for Harvester upgrades
+
+The detector accepts a node name and inferences the possible places the VMs on top of it could be live migrated to.
+If there is no place to go, it can optionally shut down the VMs.
+	`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Context(context.Background())
+		options := vmlivemigratedetector.DetectorOptions{
+			KubeConfigPath: kubeConfigPath,
+			KubeContext:    kubeContext,
+			Shutdown:       shutdown,
+			NodeName:       args[0],
+		}
+		if err := run(ctx, options); err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n", err.Error())
+			os.Exit(1)
+		}
+	},
+}
+
+func init() {
+	vmLiveMigrateDetectorCmd.Flags().BoolVar(&shutdown, "shutdown", false, "Do not shutdown non-migratable VMs")
```

Comment:
```suggestion
	vmLiveMigrateDetectorCmd.Flags().BoolVar(&shutdown, "shutdown", false, "Shut down non-migratable VMs")
```

make more sense in the help messsage:
```
Usage:
  upgrade-helper vm-live-migrate-detector NODENAME [flags]

Flags:
  -h, --help       help for vm-live-migrate-detector
      --shutdown   Do not shutdown non-migratable VMs
```